xbmc/dialogs/GUIDialogProgress.h

```diff
@@ -48,6 +48,7 @@ class CGUIDialogProgress :
    \return true if the dialog is closed, false if the user cancels early.
    */
   bool Wait(int progresstime = 10);
+  bool WaitOnEvent(CEvent& event, int progresstime);
```

Comment:
do you think progresstime should be exposed to the caller? if not, remove the parameter and set it to 10ms internally

Reply:
Maybe someone will want to tune the function's behaviour with this param. For CPluginDirectory::WaitOnScriptResult() I've set it to 20ms like other intervals.

Reply:
giving the caller the option to block the main loop is actually a bad idea. that's what this parameter does. compare with the busy dialog: it keep this parameter internally and set to 1ms